File: android/app/src/main/kotlin/com/player/tv/ui/navigation/SettingsViewModel.kt
package com.player.tv.ui.navigation

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.player.tv.data.repository.AppRepository
import com.player.tv.domain.model.Playlist
import com.player.tv.domain.model.XtreamCredentials
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch
import javax.inject.Inject

data class SettingsUiState(
    val playlists: List<Playlist> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null
)

@HiltViewModel
class SettingsViewModel @Inject constructor(
    private val repository: AppRepository
) : ViewModel() {

    private val _uiState = MutableStateFlow(SettingsUiState())
    val uiState: StateFlow<SettingsUiState> = _uiState.asStateFlow()

    init {
        loadPlaylists()
    }

    private fun loadPlaylists() {
        viewModelScope.launch {
            repository.getAllPlaylists().collect { playlists ->
                _uiState.update { it.copy(playlists = playlists) }
            }
        }
    }

    fun importM3uFromUrl(url: String, name: String) {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, error = null) }
            repository.importM3uFromUrl(url, name)
                .onSuccess {
                    _uiState.update { it.copy(isLoading = false) }
                }
                .onFailure { e ->
                    _uiState.update { it.copy(isLoading = false, error = e.message) }
                }
        }
    }

    fun importM3uFromText(content: String, name: String) {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, error = null) }
            repository.importM3uFromFile(content, name)
                .onSuccess {
                    _uiState.update { it.copy(isLoading = false) }
                }
                .onFailure { e ->
                    _uiState.update { it.copy(isLoading = false, error = e.message) }
                }
        }
    }

    fun importXtream(serverUrl: String, username: String, password: String) {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, error = null) }
            repository.importXtream(
                XtreamCredentials(serverUrl, username, password)
            ).onSuccess {
                _uiState.update { it.copy(isLoading = false) }
            }.onFailure { e ->
                _uiState.update { it.copy(isLoading = false, error = e.message) }
            }
        }
    }

    fun importStalker(serverUrl: String, macAddress: String) {
        viewModelScope.launch {
            _uiState.update { it.copy(isLoading = true, error = null) }
            repository.importStalker(serverUrl, macAddress)
                .onSuccess {
                    _uiState.update { it.copy(isLoading = false) }
                }
                .onFailure { e ->
                    _uiState.update { it.copy(isLoading = false, error = e.message) }
                }
        }
    }

    fun deletePlaylist(playlist: Playlist) {
        viewModelScope.launch {
            repository.deletePlaylist(playlist.id)
        }
    }

    fun refreshPlaylist(playlist: Playlist) {
        viewModelScope.launch {
            repository.refreshPlaylist(playlist.id)
        }
    }
}

File: ios/Features/Settings/SettingsViewModel.swift
import Foundation
import Combine

class SettingsViewModel: ObservableObject {
    @Published var playlists: [Playlist] = []
    @Published var isLoading = false
    @Published var error: String?
    
    private let repository = AppRepository()
    private var cancellables = Set<AnyCancellable>()
    
    init() {
        repository.$playlists
            .receive(on: DispatchQueue.main)
            .assign(to: &$playlists)
    }
    
    func importM3uFromUrl(_ url: String, name: String) {
        isLoading = true
        Task {
            do {
                _ = try await repository.importM3uFromUrl(url, name: name)
                isLoading = false
            } catch {
                self.error = error.localizedDescription
                isLoading = false
            }
        }
    }
    
    func importXtream(serverUrl: String, username: String, password: String) {
        isLoading = true
        Task {
            do {
                _ = try await repository.importXtream(
                    credentials: XtreamCredentials(
                        serverUrl: serverUrl,
                        username: username,
                        password: password
                    )
                )
                isLoading = false
            } catch {
                self.error = error.localizedDescription
                isLoading = false
            }
        }
    }
    
    func importStalker(serverUrl: String, macAddress: String) {
        isLoading = true
        Task {
            do {
                _ = try await repository.importStalker(
                    serverUrl: serverUrl,
                    macAddress: macAddress
                )
                isLoading = false
            } catch {
                self.error = error.localizedDescription
                isLoading = false
            }
        }
    }
    
    func deletePlaylist(_ playlist: Playlist) {
        repository.deletePlaylist(playlist)
    }
    
    func refreshPlaylist(_ playlist: Playlist) {
        Task {
            try? await repository.refreshPlaylist(playlist)
        }
    }
}
